apisix/upstream.lua

```diff
@@ -81,7 +81,11 @@ end
 _M.set = set_directly
 
 
-local function release_checker(healthcheck_parent)
+local function release_checker(healthcheck_parent, is_compacted)
+    if is_compacted then
```

Comment:
This is a wrong solution and will cause the checker to never be released

Reply:
could you clarify more? My changes will not release the checker only when the reason is compaction. There is still another case that can release the checkers properly:

https://github.com/shreemaan-abhishek/apisix/blob/105993f40f34768cd0a86035d01393346fff3a94/apisix/core/config_etcd.lua#L686